src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,19 +54,22 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block.getBlock() + "," + street.getStreet() + "," + unit.getUnit() + "," + postal.getPostal();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.equals(((Address) other).block))
+                && this.street.equals(((Address) other).street)
+                && this.unit.equals(((Address) other).unit)
+                && this.postal.equals(((Address)other).postal); // state check
     }
```

Comment:
Good job in modifying the .equals method correspondingly to how the variables are handled. However, I believe when using .equals for those, you will have to override it in each class as well.